### PR TITLE
Jsx max props per line updates

### PR DESCRIPTION
**Pull request by kentor**
addresses parts of https://github.com/yannickcr/eslint-plugin-react/issues/878

will happily change option names

### CODE REVIEW

File: tests/lib/rules/jsx-max-props-per-line.js [hunk 1]
```diff
@@ -89,5 +100,24 @@ ruleTester.run('jsx-max-props-per-line', rule, {
     ].join('\n'),
     errors: [{message: 'Prop `this.props` must be placed on a new line'}],
     parserOptions: parserOptions
+  }, {
+    code: [
+      '<App',
+      '  foo={{',
```
Comment (ljharb) [inline tests/lib/rules/jsx-max-props-per-line.js]: Let's also include some tests where the tagname and first prop are on the same line, but there's another prop. Also, please add variants of all tests that include spread props

Comment (kentor) [inline tests/lib/rules/jsx-max-props-per-line.js]: sure thing

**Comment by revisionfour** [comment]
I think this is ready to be merged and should be merged.

File: docs/rules/jsx-max-props-per-line.md [hunk 2]
```diff
@@ -56,6 +60,28 @@ The following patterns are not considered warnings:
...
 
+### `when`
+
+Possible values:
+- `always` (default) - Always check for max props per line.
+- `multiline` - Only check for max props per line when jsx tag spans multiple lines.
+
+The following patterns are considered warnings:
+```jsx
+// [1, {when: always}]
```
Comment (lencioni) [inline docs/rules/jsx-max-props-per-line.md]: Should this be

```js
// [1, {when: 'always'}]
```

Comment (kentor) [inline docs/rules/jsx-max-props-per-line.md]: Not sure. It's valid if we're talking yaml, but not if it's json/js format.

Comment (ljharb) [inline docs/rules/jsx-max-props-per-line.md]: I think json format is the most common by a massive amount, so that's the format we should stick to in docs.

**Comment by princed** [comment]
Any chance to get this merged?

**Pull request merged**